shop/src/main/java/ro/msg/learning/shop/Entities/Order.java

```diff
@@ -0,0 +1,59 @@
+package ro.msg.learning.shop.Entities;
+
+import lombok.*;
+import org.hibernate.annotations.Fetch;
+import org.hibernate.annotations.FetchMode;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Table(name = "Order_")
+@Builder
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @EqualsAndHashCode.Include
+    private Integer id;
+
+    @ManyToMany
+    @JoinTable(
+            name = "order_locations",
+            joinColumns = @JoinColumn(name = "order_id"),
+            inverseJoinColumns = @JoinColumn(name = "location_id")
+    )
+    @Fetch(value = FetchMode.SELECT)
+    @ToString.Exclude
+    private Set<Location> shippedFrom;
+
+    @ManyToOne
+    private Customer customer;
+
+    @EqualsAndHashCode.Include
```

Comment:
seems that you included more than excluded fields. could use  @EqualsAndHashCode(exclude="XXYZ"), but what's the reason behind excluding those fields?